client/hotrod-client/src/main/java/org/infinispan/client/hotrod/RemoteCache.java

```diff
@@ -343,4 +344,9 @@
     * Executes a remote script passing a set of named parameters
     */
    <T> T execute(String scriptName, Map<String, ?> params);
+
+   /**
+    * Returns segment ownership for this cache
+    */
+   Map<SocketAddress, Set<Integer>> getSegmentsByServer();
```

Comment:
> It is definitely not a user-oriented API

Yeah, it's like a SPI, maybe an advanced API :smile:

One of the ```Remotecache.retrieveEntries``` accepts a list of segments to filter, IMO it'd be a bit odd to hide ```getSegmentsByServer``` since it's a useful complement. We discussed previously the possibility of a ```AdvancedRemoteCache``` interface, it's not too late to rescue that idea if there's strong opposition to expose segments & co in the ```RemoteCache```

> we have to keep it stable

Not trying to predict the future, but segment ownership by servers is a solidified concept that is not likely to go away anytime soon?